### PR TITLE
issue #276 fixed (validation for cluster name, environment name, namespace)

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
…space)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/cluster/Cluster.tsx [hunk 1]
```diff
@@ -222,7 +222,13 @@ function ClusterForm({ id, cluster_name, server_url, active, config, environment
         {
             cluster_name: {
                 required: true,
-                validator: { error: 'Name is required', regex: /^.*$/ }
+                validators: [
+                                { error: 'Name is required',  regex: /^.*$/ },
+                                { error: "Use only lowercase alphanumeric characters, '-' or '.'", regex: /^[a-z0-9_\.]+$/ },
+                                { error: "Start/End with an alphanumeric character", regex: /^[a-z0-9](.*[a-z0-9])?$/ },
+                                { error: "Do not use spaces, '-' or '.'", regex: /^\S+$/ },
```
Comment (pghildiyal) [inline src/components/cluster/Cluster.tsx:229]: it only checks for space, but this ll be caught by 227

File: src/components/cluster/Cluster.tsx [hunk 2]
```diff
@@ -390,11 +396,23 @@ function Environment({ environment_name, namespace, id, cluster_id, handleClose,
         {
             environment_name: {
                 required: true,
-                validator: { error: 'This is required field(max 16 chars).', regex: /^.{1,16}$/ }
+                validators: [
+                    { error: 'Environment name is required',  regex: /^.*$/ },
+                    { error: "Use only lowercase alphanumeric characters, '-' or '.'", regex: /^[a-z0-9_\.]+$/ },
+                    { error: "Start/End with an alphanumeric character", regex: /^[a-z0-9](.*[a-z0-9])?$/ },
+                    { error: "Do not use spaces, '-' or '.'", regex: /^\S+$/ },
+                    { error: "Minimum 5 and Maximum 25 characters required", regex: /^.{5,25}$/ }
```
Comment (pghildiyal) [inline src/components/cluster/Cluster.tsx:404]: it only checks for space, but this ll be caught by 401

File: src/components/cluster/Cluster.tsx [hunk 3]
```diff
@@ -390,11 +396,23 @@ function Environment({ environment_name, namespace, id, cluster_id, handleClose,
...
+                ]
             },
             namespace: {
                 required: isNamespaceMandatory,
-                validator: { error: '^[a-z]+[a-z0-9\-\?]*[a-z0-9]+$ pattern should satisfy.', regex: /^[a-z]+[a-z0-9\-\?]*[a-z0-9]+$/ }
+                validators: [
+                    { error: 'Namespace is required',  regex: /^.*$/ },
+                    { error: "Use only lowercase alphanumeric characters, '-' or '.'", regex: /^[a-z0-9_\.]+$/ },
+                    { error: "Start/End with an alphanumeric character", regex: /^[a-z0-9](.*[a-z0-9])?$/ },
+                    { error: "Do not use spaces, '-' or '.'", regex: /^\S+$/ },
```
Comment (pghildiyal) [inline src/components/cluster/Cluster.tsx:413]: it only checks for space, but this ll be caught by 411

File: src/components/cluster/Cluster.tsx (hunk 1 above)
Comment (manish-agrawal-ai) [reply src/components/cluster/Cluster.tsx:229]: correct

File: src/components/cluster/Cluster.tsx (hunk 2 above)
Comment (manish-agrawal-ai) [reply src/components/cluster/Cluster.tsx:404]: correct

File: src/components/cluster/Cluster.tsx (hunk 3 above)
Comment (manish-agrawal-ai) [reply src/components/cluster/Cluster.tsx:413]: correct

File: src/components/common/helpers/Helpers.tsx [hunk 4]
```diff
@@ -84,13 +84,31 @@ export function useForm(stateSchema, validationSchema = {}, callback) {
...
             }
-        } else {
         }
+
+        // multiple validators
+        let _validators =  validationSchema[name].validators
+        if (_validators && typeof _validators === 'object' && Array.isArray(_validators)) {
+            _validators.forEach((_validator) => {
+                if(!validateSingleValidator(_validator, value)){
+                    return _validator.error;
```
Comment (pghildiyal) [inline src/components/common/helpers/Helpers.tsx:106]: this should be set in error

Comment (manish-agrawal-ai) [reply src/components/common/helpers/Helpers.tsx:106]: this is return of validateField function

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**